kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -118,15 +120,16 @@ public PartitionSegments(
           createSegment(segmentToCreate);
         }
 
-        final long firstSegmentId = activeSegments.get(0).segmentId;
+        final long firstSegmentId = activeSegments.get(0).segmentStartTimestamp;
```

Comment:
nit: change variable name to `segmentStartTimestamp`

We'll also need to update the way the ranges are printed since `firstSegmentId + activeSegments.size()` only makes sense when talking about segment ids. See suggestion below